source/code/troubleshooter/modules/errors_tsg.py

```diff
@@ -25,19 +26,12 @@
 
 
 
-# get error codes from Troubleshooting.md
-def get_error_codes(err_type):
-    # extension
-    if (err_type == 'Extension'):
-        return (extension_err_codes_dict, None)
-
-    # shell bundle or onboarding
-    err_codes = {0 : "No errors found"}
-    section_name = "{0} Error Codes".format(err_type)
+# parse through error codes to get dictionary
+def parse_error_codes(ts_doc, section_name):
     try:
-        ts_doc = urlopen(TSG_URL)
+        err_codes = {0 : "No errors found"}
         section = None
-        for line in ts_doc.readlines():
+        for line in ts_doc:
             line = line.decode('utf8').rstrip('\n')
```

Comment:
does this work both py2/3?

Reply:
Sorry yes you're right, I accidentally left that in after I changed Troubleshooting.md from just a urlopen check to checking both with the file and urlopen. I changed it so it only tries to decode if it's opening the file via url, which works on both Python2 and Python3 in both file and url form:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/2da374a1507036bbd4a730dbd4f7f9ced60843be/source/code/troubleshooter/modules/errors_tsg.py#L35-L37